syntax = "proto3";  // プロトコルバッファのバージョンを指定

option go_package = "api-go/src/service/gateway";

// GetStockServiceというサービスを定義
service GetStockService {
  // GetStockDataというRPCメソッドを定義
  rpc GetStockData (GetStockRequest) returns (GetStockResponse) {}
}

// リクエストメッセージを定義
message GetStockRequest {
  string ticker = 1;  // 銘柄コードを格納するフィールド
}

// レスポンスメッセージを定義
message GetStockResponse {
  map<string, double> stock_data = 1;  // 株価データを含むフィールド
  string date = 2;                     // 日付を含むフィールド
}

// python用コンパイルコマンド
// python -m grpc_tools.protoc -I src/get_stock --python_out src/get_stock --grpc_python_out=src/get_stock src/get_stock/get_stock.proto

// Go用コンパイルコマンド
// protoc --proto_path=../data-analysis-python/src/get_stock --go_out=./src/service/gateway --go_opt=paths=source_relative --go-grpc_out=./src/service/gateway --go-grpc_opt=paths=source_relative ../data-analysis-python/src/get_stock/get_stock.proto